# Loading PI1M Database
* Reference) Ma, Ruimin, and Tengfei Luo. "PI1M: a benchmark database for polymer informatics." *Journal of Chemical Information and Modeling* 60.10 (**2020**): 4684-4690.

In [1]:
import pandas as pd

df = pd.read_csv('../PI1M.csv')
df

SMILES
0                       *CCC[Fe]CCCC(=O)OCCCCOCCCNCC(*)=O
1                              *CCCC1C=CNC2=CC=C2C(*)CCC1
2                                                   *CCC*
3                               *C(=O)CNC(*)C(=O)OCCCCCNC
4                              *CC(C)(C)CCCCCCC(C)C(=O)N*
...                                                   ...
995794  *CC(*)(CCC(=O)Oc1ccc(/N=N/c2ccc(C#N)cc2)cc1)c1...
995795  *Oc1ccccc1NC(=O)c1ccc(/N=N/c2ccc(C(=O)OC(=O)CC...
995796                    *NC(CC)CC(=O)NCCCCCCNS(*)(=O)=O
995797            *CCN(CCOC(=O)NCCCCCCCCNC(=O)O*)CC(=O)OC
995798               *OP(=O)(Cl)c1ccc(C(O)c2ccc(*)cc2)cc1

[995799 rows x 1 columns]

PI1M은 995,799 개의 가상 고분자 SMILES string 을 포함하고 있다.

## Canonical SMILES

In [2]:
import rdkit
from rdkit import Chem

In [3]:
def canonical_smiles(smiles):
    return Chem.CanonSmiles(smiles)

In [4]:
%%time
df['Canonical_SMILES'] = df['SMILES'].apply(canonical_smiles)

CPU times: user 5min 55s, sys: 243 ms, total: 5min 55s
Wall time: 5min 55s


In [5]:
False in (df['SMILES']==df['Canonical_SMILES'])

True

In [6]:
df.to_csv('canonical_PI1M.csv')

어떤 화학구조의 SMILES 를 unique representation 으로 만드는 canonicalization 알고리즘은 DB마다 다를 수 있다.
RDKit 기준으로는 canonical smiles 가 아닌 smiles 가 포함되어 있다.